check-update-security.sh

```diff
@@ -0,0 +1,230 @@
+    exit 1
+fi
+
+# Get ecs-optimized ami's PATH in ssm parameters
+platform=$1
+skip_user_data=0
+instance_type="c5.large"
+install_and_start_ssm_agent=0
+case "$platform" in
+"al1")
+    ami_path=$AL1_PATH
+    install_and_start_ssm_agent=1 # AL1 Requires Installation of SSM Agent
+    ;;
+"al2")
+    ami_path=$AL2_PATH
+    ;;
+"al2_arm")
+    ami_path=$AL2_ARM_PATH
+    instance_type="c6g.medium"
+    ;;
+"al2023")
+    ami_path=$AL2023_PATH
+    skip_user_data=1
+    ;;
+"al2023_arm")
+    ami_path=$AL2023_ARM_PATH
+    skip_user_data=1
+    instance_type="c6g.medium"
+    ;;
+*)
+    error_msg "Incorrect platform selection"
+    usage
+    exit 1
+    ;;
+esac
+
+# Query ssm to get latest ecs optimized ami
+ami_id=$(aws ssm get-parameters --names $ami_path --region us-west-2 | jq -r '.Parameters[0].Value' | jq -r '.image_id')
+
+user_data=$(mktemp user_data.txt)
+if [ "$install_and_start_ssm_agent" -eq 1 ]; then
+
+    cat <<EOT >>user_data.txt
+Content-Type: multipart/mixed; boundary="//"
+MIME-Version: 1.0
+
+--//
+Content-Type: text/cloud-config; charset="us-ascii"
+MIME-Version: 1.0
+Content-Transfer-Encoding: 7bit
+Content-Disposition: attachment; filename="cloud-config.txt"
+
+#cloud-config
+repo_upgrade: none
+
+--//
+Content-Type: text/x-shellscript; charset="us-ascii"
+MIME-Version: 1.0
+Content-Transfer-Encoding: 7bit
+Content-Disposition: attachment; filename="userdata.txt"
+
+#!/bin/bash
+cd /tmp
+sudo yum install -y https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_amd64/amazon-ssm-agent.rpm
+sudo start amazon-ssm-agent
+--//--
+EOT
+
+else
+    echo "#cloud-config" >>user_data.txt
+    echo "repo_upgrade: none" >>user_data.txt
+fi
+
+# Launch ec2 instance with given ami and SSM access for command execution
+# Also get instance id
+if [ $skip_user_data -eq 0 ]; then
+    # Modify user data to ignore automatic updates by al and al2
+    instance_id=$(aws ec2 run-instances \
+        --image-id $ami_id \
+        --instance-type $instance_type \
+        --iam-instance-profile Arn=$IAM_INSTANCE_PROFILE_ARN \
+        --user-data file://user_data.txt |
+        jq -r '.Instances[0].InstanceId')
+    command_params='commands=["yum check-update --security --sec-severity=critical  --exclude=nvidia*,docker*,cuda*,containerd* -q"]'
+else
+    # In the case that we
+    instance_id=$(aws ec2 run-instances \
+        --image-id $ami_id \
+        --instance-type $instance_type \
+        --iam-instance-profile Arn=$IAM_INSTANCE_PROFILE_ARN |
+        jq -r '.Instances[0].InstanceId')
+    distribution_release_al2023=$(curl -s https://al2023-repos-us-west-2-de612dc2.s3.dualstack.us-west-2.amazonaws.com/core/releasemd.xml |
+        xmllint --xpath "string(//root/releases/release[last()]/@version)" -)
+    command_params='commands=["sudo dnf check-update --releasever='$distribution_release_al2023' --security --sec-severity=Critical --exclude=nvidia*,docker*,cuda*,containerd* -q"]'
```

Comment:
nit: We are not checking a GPU AMI, are `nvidia*` and `cuda*` here necessary?

Reply:
Or actually should we consider GPU AMIs too

Reply:
I believe this is due to these packages being pinned.